File: Juan/P4/Java/Napakalaki/src/NapakalakiGame/CultistPlayer.java

package NapakalakiGame;

import java.util.Random;

/**
 *
 * @author Raul
 */
public class CultistPlayer extends Player {
    
    /* Atributo */
    private static int totalCultistPlayers = 0;
    
    /* Relación */
    private Cultist myCultistCard;
    
    /*Constructor */
    
    /**
     * CultisPlayer(Player p, Cultist c)
     * @param p de tipo Player
     * @param c de tipo Cultist
     */
    CultistPlayer(Player p, Cultist c){
        super(p);
        this.myCultistCard = c;
        this.totalCultistPlayers = this.totalCultistPlayers + 1;
    }
    
    /**
     * Método getCombatLevel()
     * Devuelve la suma del combate
     * @return suma
     */
    
    @Override
    protected int getCombatLevel() {
        int suma;
        double veintexciento;
        
        veintexciento = super.getCombatLevel()*0.2;
        
        suma = super.getCombatLevel() + (int)veintexciento + myCultistCard.getGainedLevels();
        return suma;
    }
    
    /**
     * Método getOpenentLevel(Monster m)
     * Devuelve el nivel del oponente
     * @param m Monster
     * @return int
     */
    @Override
    protected int getOponentLevel(Monster m){
        Monster aux = m;
        return aux.getCombatLevelAgainstCultistPlayer();
    }
    
    /**
     * Método  shouldConvert()
     * Se obtiene el valor boolean de la clase player de
     * lanzar el dado y devolverá true si
     * se obtiene un 1 y false en caso contrario
     * @return boolean
     */
    @Override
    protected boolean shouldConvert(){        
        return  super.shouldConvert();
    }
    /**
     * Método giveMeAtreasure()
     * Devuelve un tesoro
     * @return 
     */
    @Override
     protected Treasure giveMeAtreasure(){
        Random rand = new Random();
        int numero = rand.nextInt(super.visibleCultist.size()-1); 
        
        return  super.visibleCultist.get(numero);
    }
    /**
     * Método canYouGiveMeAtreasure()
     * Devuelve true si tiene tesoros visibles
     * el enemigo, false en caso contrario
     * @return true or false
     */
    @Override
    protected boolean canYouGiveMeAtreasure(){
        if(!super.enemy.getVisibleTreasures().isEmpty())
            return true;
        
        else
           return false;
           
    }
    /**
     * Método getTotalCultistPlayers()
     * Devuelve el número total de CultistPlayers
     * @return totalCultistPlayers int
     */
    public static int getTotalCultistPlayers() {
        return totalCultistPlayers;
    }
    
  
    
    
    
    
}

File: Juan/P4/Ruby/Napakalaki/lib/CultistPlayer.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'Treasure.rb'
require_relative 'Monster.rb'
require_relative 'TreasureKind.rb'
require_relative 'CombatResult.rb'
require_relative 'Dice.rb'
require_relative 'BadConsequence.rb'
require_relative 'Player.rb'



module NapakalakiGame
class CultistPlayer < Player   
     
    @@totalCultistPlayers = 0
    @myCultistCard = Cultist.new('nada',0)
    
    def initialize(p,c)
        @name = p.name
        @level = p.level
        @dead = p.dead
        @canISteal = p.canISteal
        @enemy = p.enemy

        @pendingBadConsequence = p.pendingBadConsequence

        @hiddenTreasures = p.hiddenTreasures
        @visibleTreasures = p.visibleTreasures
        @myCultistCard = c
        
        incTotal
    end 
    
    def self.incTotal
        @@totalCultistPlayers = @@totalCultistPlayers + 1
    end
    
    protected
    def getCombatLevel
        
        number = super.getCombatLevel *0.2
        
        
        return super.getCombatLevel +number + @ic
    end
    
    def getOponentLevel(m)
        m.getCombatLevelAgainstCultistPlayer
    end
    
    def shouldConvert
        return false
    end
    
    
    private
    def giveMeATreasure
        number = rand(@visibleTreasures.size-1)
        
        return @visibleTreasures[number]
    end
    
    protected
    def canYouGiveMeATreasure
        if (!enemy.getVisibleTreasures.empty?)
            return true
        else
            return false
        end
    end
    
    public
    def getTotalCultistPlayers
        
    end
     
    
end
end